syfertext/doc.py

```diff
@@ -34,6 +35,26 @@ def __init__(
         # file
         self.container = list()
 
+        # Initialize the Underscore object (inspired by spaCy)
+        # This object will hold all the custom attributes set
+        # using the `self.set_attribute` method
+        self._ = Underscore()
+
+    def set_attribute(self, name: str, value: object):
+        """Creates a custom attribute with the name `name` and
+           value `value` in the Underscore object `self._`
+
+        Args:
+            self (Doc): current Doc.
+            name (str): name of the custom attribute .
+            value (object): value of the custom named attribute.
+        """
+
+        # make sure there is no space in name as well prevent empty name
+        assert len(name) > 0 and (not (" " in name))
+
```

Comment:
Please add an assert for the type of `name` as `str`
Also, please add a clear message to the assert to describe the error

Reply:
Please use isinstance(name, str) instead. It should come as the first condition. 

The error message sure be 

"Argument `name` should be a non-empty  `str` type containing no spaces"